src/client/Microsoft.Identity.Client/Instance/Authority.cs

```diff
@@ -36,22 +27,65 @@ protected Authority(AuthorityInfo authorityInfo)
         public AuthorityInfo AuthorityInfo { get; }
 
         #region Builders
-        public static Authority CreateAuthorityWithOverride(AuthorityInfo requestAuthorityInfo, AuthorityInfo configAuthorityInfo)
+
+        /// <summary>
+        /// Figures out the authority based on the authority from the config and the authority from the request, 
+        /// and optionally the homeAccountTenantId, which has an impact on AcquireTokenSilent
+        /// 
+        /// The algorithm is: 
+        /// 
+        /// 1. If there is no request authority (i.e. no authority override), use the config authority. 
+        ///     1.1. For AAD, if the config authority is "common" etc, try to use the tenanted version with the home account tenant ID
+        /// 2. If there is a request authority, try to use it. 
+        ///     2.1. If the request authority is not "common", then use it
+        ///     2.2  If the request authority is "common", ignore it, and use 1.1
+        /// 
+        /// </summary>
+        /// <param name="configAuthorityInfo"></param>
+        /// <param name="requestAuthorityInfo"></param>
+        /// <param name="requestHomeAccountTenantId"></param>
+        /// <returns></returns>
+        public static Authority CreateAuthorityForRequest(
+            AuthorityInfo configAuthorityInfo,
+            AuthorityInfo requestAuthorityInfo,
+            string requestHomeAccountTenantId = null)
         {
-            switch (requestAuthorityInfo.AuthorityType)
+            if (configAuthorityInfo == null)
+            {
+                throw new ArgumentNullException(nameof(configAuthorityInfo));
+            }
+
+            switch (configAuthorityInfo.AuthorityType)
             {
+                // ADFS and B2C are tenantless, no need to consider tenant 
```

Comment:
>B2C are tenantless [](start = 28, length = 18)

What you mean by tenantless here? B2C has a tenant.

Reply:
From MSAL perspective, I don't think it does. If you use .WithTentantId() or overloads of WithAuthority() that take a tenant, I don't think it does anything to a B2C authority. 

Tenantless means "common", "organizations" or "consumers". Not sure how else to call it ... "commonish" ? :)

Reply:
idk...but definitely don't say "commonish"

Reply:
not-tenanted authorties?